### ISSUE TITLE
Rule `aria-roles` should return inapplicable when role is undefined, empty or whitespace

### ISSUE THREAD

**Issue by jkodu** [dequelabs/axe-core#2074]
Expectation: 
`aria-roles` should return inapplicable when role is undefined, empty or whitespace

Actual: 
`aria-roles` currently returns a failure.

Motivation:
Outcome of such examples in ACT Rules - https://act-rules.github.io/rules/674b10#inapplicable-example-1 where inapplicable is expected is returned as failure instead.

**Comment by padmavemulapati** [comment]
Verified with axe-core code and as well latest axe-coconut and axe , I am seeing the fail, if  aria-role  is not given (empty, whitespace or undefined)
![image](https://user-images.githubusercontent.com/47409579/75983342-619bc200-5f0e-11ea-8b5a-44f229a33f8c.png)
![image](https://user-images.githubusercontent.com/47409579/75983484-a6275d80-5f0e-11ea-8cf1-ce53f9573a87.png)

**Comment by jkodu** [comment]
@padmavemulapati this is a fix in axe-core. I am unsure if the fix got propagated to axe coconut yet. Please test this with in axe-core.

Integration tests were added for this fix - https://github.com/dequelabs/axe-core/pull/2077/files#diff-edc8f954a673ea2fa1829a93d566f548

**Comment by padmavemulapati** [comment]
Verified with axe.js@3.5.1 version, from spource snippet seeing the inapplicable aria-role 
![image](https://user-images.githubusercontent.com/47409579/76213670-2ad8eb00-6231-11ea-9e01-8d0c95102ed8.png)

**Comment by jkodu** [comment]
Nothing to document here. Closing issue.
